src/GitVersionCore.Tests/ConfigProviderTests.cs

```diff
@@ -29,46 +29,26 @@ public void Setup()
     }
 
     [Test]
-    public void CanReadDocumentAndMigrate()
+    public void CanReadOldDocument()
     {
         const string text = @"
-assembly-versioning-scheme: MajorMinor
-next-version: 2.0.0
-tag-prefix: '[vV|version-]'
-mode: ContinuousDelivery
+assemblyVersioningScheme: MajorMinor
+develop-branch-tag: alpha
+release-branch-tag: rc
 branches:
-    develop:
+    dev(elop)?(ment)?$:
         mode: ContinuousDeployment
         tag: dev
     release[/-]:
        mode: continuousDeployment
-       tag: rc 
-";
-        SetupConfigFileContent(text);
-
-        var config = ConfigurationProvider.Provide(repoPath, fileSystem);
-        config.AssemblyVersioningScheme.ShouldBe(AssemblyVersioningScheme.MajorMinor);
-        config.AssemblyInformationalFormat.ShouldBe(null);
-        config.NextVersion.ShouldBe("2.0.0");
-        config.TagPrefix.ShouldBe("[vV|version-]");
-        config.VersioningMode.ShouldBe(VersioningMode.ContinuousDelivery);
-        config.Branches["dev(elop)?(ment)?$"].Tag.ShouldBe("dev");
-        config.Branches["releases?[/-]"].Tag.ShouldBe("rc");
-        config.Branches["releases?[/-]"].VersioningMode.ShouldBe(VersioningMode.ContinuousDeployment);
-        config.Branches["dev(elop)?(ment)?$"].VersioningMode.ShouldBe(VersioningMode.ContinuousDeployment);
-    }
-
-    [Test]
-    public void CanReadOldDocument()
-    {
-        const string text = @"
-assemblyVersioningScheme: MajorMinor
-develop-branch-tag: alpha
-release-branch-tag: rc
+       tag: rc
 ";
         SetupConfigFileContent(text);
         var error = Should.Throw<OldConfigurationException>(() => ConfigurationProvider.Provide(repoPath, fileSystem));
         error.Message.ShouldContainWithoutWhitespace(@"GitVersion configuration file contains old configuration, please fix the following errors:
+GitVersion branch configs no longer are keyed by regexes, update:
```

Comment:
I put a bit of effort into the error messages to give a good upgrade experience.